src/NServiceBus.Core/obsoletes.cs

```diff
@@ -1029,12 +1071,90 @@ public interface IWantToRunWhenConfigurationIsComplete
     }
 
     [ObsoleteEx(
+         Message = Error,
+         RemoveInVersion = "7",
+         TreatAsErrorFromVersion = "6")]
+    public class SecondLevelRetriesConfig : ConfigurationSection
+    {
+        const string Error = "Second Level Retries has been renamed to Delayed Retries. The app.config API has been removed, use the code API via endpointConfiguration.Recoverability().Delayed(settings => ...);.";
+
+        [ObsoleteEx(
+             Message = Error  + " To disable use endpointConfiguration.Recoverability().Delayed(settings => settings.NumberOfRetries(0));",
+             RemoveInVersion = "7",
+             TreatAsErrorFromVersion = "6")]
+        public bool Enabled
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        [ObsoleteEx(
+             Message = Error + " To change the TimeIncrease use endpointConfiguration.Recoverability().Delayed(settings => settings.TimeIncrease(TimeSpan.FromMinutes(5));",
+             RemoveInVersion = "7",
+             TreatAsErrorFromVersion = "6")]
+        public TimeSpan TimeIncrease
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        [ObsoleteEx(
+             Message = Error + " To change the NumberOfRetries use endpointConfiguration.Recoverability().Delayed(settings => settings.NumberOfRetries(5);",
+             RemoveInVersion = "7",
+             TreatAsErrorFromVersion = "6")]
+        public int NumberOfRetries
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+    }
+
+    [ObsoleteEx(
         TreatAsErrorFromVersion = "6",
         RemoveInVersion = "7",
         ReplacementTypeOrMember = "EndpointConfiguration.EnlistWithLegacyMSMQDistributor")]
     public class MasterNodeConfig : ConfigurationSection
     {
     }
+
+    [ObsoleteEx(
+        Message = Error,
+        RemoveInVersion = "7",
+        TreatAsErrorFromVersion = "6")]
+    public class TransportConfig : ConfigurationSection
+    {
+        const string Error = "The app.config API TransportConfig has been removed, use the code API.";
+
+        [ObsoleteEx(
+            TreatAsErrorFromVersion = "6",
+            RemoveInVersion = "7",
+            Message = Error + " To change the concurrency level use endpointConfiguration.LimitMessageProcessingConcurrencyTo(1);")]
+        public int MaximumConcurrencyLevel
+        {
+            get { throw new NotImplementedException(); }
```

Comment:
This would mean that users will end up with a NotImplementedException when upgrading without a further hint how to resolve this when using this in the app.config or do I miss something? That was also the reason why we have `DetectObsoleteConfigurationSettings`, to give users a better obsolete message for obsoleted xml sections because there aren't any compilation errors.